### PR TITLE
Adding new API for JSON Mirror function

### PR DESCRIPTION
**Pull request by mwang87**

### CODE REVIEW

File: run_server.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 source activate usi
 
-gunicorn -w 2 -b 0.0.0.0:5000 --timeout 3600 main:app --access-logfile /app/logs/access.log --max-requests 1000
+gunicorn -w 2 --threads=1 -b 0.0.0.0:5000 --timeout 3600 main:app --access-logfile /app/logs/access.log --timeout 30 --max-requests 500 --max-requests-jitter 100
```
Comment (bittremieux) [inline run_server.sh:4]: Double `timeout` argument.

File: views.py [hunk 2]
```diff
@@ -714,6 +714,45 @@ def peak_json():
...
+        spectrum1, source1, splash_key1 = parsing.parse_usi(usi1)
+        spectrum2, source2, splash_key2 = parsing.parse_usi(usi2)
+        spectrum1, spectrum2 = _prepare_mirror_spectra(spectrum1, spectrum2,
+                                                       plotting_args)
+        similarity, peak_matches = _cosine(
+            spectrum1, spectrum2, plotting_args['fragment_mz_tolerance'],
+            plotting_args['cosine'] == 'shifted')
+
+        spectrum1_dict = {
+            'peaks': _get_peaks(spectrum1),
```
Comment (bittremieux) [inline views.py:733]: Now the peaks are as after spectrum processing (m/z range restriction and base peak normalization). Do we want to return the unmodified spectra instead? That's what happens with the single spectrum JSON API.

File: views.py [hunk 3]
```diff
@@ -714,6 +714,45 @@ def peak_json():
...
+        spectrum2_dict = {
+            'peaks': _get_peaks(spectrum2),
+            'n_peaks': len(spectrum2.mz),
+            'precursor_mz': spectrum2.precursor_mz,
+            'splash': splash_key2
+        }
+        result_dict = {'spectrum1': spectrum1_dict,
+                       'spectrum2': spectrum2_dict,
+                       'cosine': similarity,
+                       'peak_matches_count': len(peak_matches),
```
Comment (bittremieux) [inline views.py:747]: Maybe for consistency use `n_peak_matches` instead? For the JSON endpoint we use as key `n_peaks` for the number of peaks in a spectrum.

Comment (mwang87) [reply views.py]: updated.

File: views.py (hunk 2 above)
Comment (mwang87) [reply views.py:734]: Fixed.

File: run_server.sh (hunk 1 above)
Comment (mwang87) [reply run_server.sh]: set to lower one.

**Comment by mwang87** [comment]
@bittremieux ready to look at again!

**Comment by bittremieux** [comment]
Test are failing because of timeouts and retrieval issues, so we have to look into that. The next functionality works though so I'm merging this PR.

**Pull request merged**